### PR TITLE
AVRO-1931: Reader is now compatible if able to read all branches of u…

### PR DESCRIPTION
**Pull request by epkanol**
…nion

### CODE REVIEW

**Comment by epkanol** [comment]
New try, this time forking apache/avro on github

**Comment by harshach** [comment]
+1.

**Comment by harshach** [comment]
@rdblue we are also looking to have these patches in Avro. Appreciate if committers can take a look.

File: lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
+package org.apache.avro;
```
Comment (spacharya) [inline lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java]: Missing headers for the apache license

File: lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+public class TestReadingWritingDataInEvolvedSchemas {
+
+  private static final String RECORD_A = "RecordA";
+  private static final String FIELD_A = "fieldA";
+  private static final char LATIN_SMALL_LETTER_O_WITH_DIARESIS = '\u00F6';
+  
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+  //
```
Comment (spacharya) [inline lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java]: not sure why ```//``` present on multiple lines.

File: lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java (hunk 1 above)
Comment (epkanol) [inline lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java]: Added ASL (copied from other test class). Updated same commit (amend, push -f)

**Comment by harshach** [comment]
@epkanol can you please addres @spacharya review. Thanks.

File: lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java (hunk 1 above)
Comment (epkanol) [inline lang/java/avro/src/test/java/org/apache/avro/TestReadingWritingDataInEvolvedSchemas.java]: @harshach my branch (actually, the same commit) was updated 10th of April, don't know what else to do. Waiting for your review and acceptance/merge into master.
The Apache License was added to my new file, and the reason the // is there in the file is in order to preserve formatting across IDEs (My Eclipse has been acting erratically lately - Neon upgrade issue)

**Comment by busbey** [comment]
@harshach thanks for reviewing this! What Author string would you like used for a signed-off-by line?

**Pull request closed without merging**